### PR TITLE
Fix disconnection

### PR DESCRIPTION
**Pull request by yatharthranjan**
- The FCM server sends disconnection requests once in a while (according to docs this is for periodic maintenance, cleaning of stale connections, etc). When this occurs the connection needs to be closed and connected again.
- The smack library sends a `presence unavailable` (`<presence id='tMZDi-119397' type='unavailable'></presence>`) message and closes the connection when such a request is received from the server as mentioned in the java docs in this PR.
- This PR adds an interceptor to intercept the presence message and issue a reconnection.

- This also increases the log retention from 7 days to 30 days for normal level logs and 60 days for error logs.

### CODE REVIEW

File: src/main/java/com/wedevol/xmpp/server/CcsClient.java [hunk 1]
```diff
@@ -487,7 +504,7 @@ public void connectionClosedOnError(Exception e) {
     @Override
     public void connectionClosed() {
         logger.info("Connection closed. The current connectionDraining flag is: {}", isConnectionDraining);
-        if (isConnectionDraining) {
+        if (isConnectionDraining || isPresenceUnavailable) {
```
Comment (afolarin) [inline src/main/java/com/wedevol/xmpp/server/CcsClient.java]: do we need a back-off on reconnect?

Comment (yatharthranjan) [reply src/main/java/com/wedevol/xmpp/server/CcsClient.java]: It only backs-off every time it cannot connect. So in this case it should connect immediately but will back-off if it cannot.

**Comment by yatharthranjan** [comment]
> Might be good to keep a monitor on the logs for disconnection (in case there are any other scenarios where this happens)
Yes that is exactly how I found this bug, as this is very internal and no error is thrown since its the expected behaviour but apparently is not documented well anywhere.

**Comment by afolarin** [comment]
ok great.

> expected behaviour but apparently is not documented well anywhere.
:thinking:  you might want to send a PR upstream too

**Comment by yatharthranjan** [comment]
> 🤔 you might want to send a PR upstream too

Yes perhaps. But our codebase includes various things specific to Radar-base.

**Pull request merged**